src/network/p2p_topology.rs

```diff
@@ -1,22 +1,74 @@
 //! module defining the p2p topology management objects
 //!
 
+use bincode;
+use chain_core::property;
+use network_core::gossip;
 use poldercast::topology::{Cyclon, Module, Rings, Topology, Vicinity};
 use poldercast::Subscription;
 pub use poldercast::{Address, Id, InterestLevel, Node};
 use std::collections::BTreeMap;
-
 use std::sync::{Arc, RwLock};
 
 pub const NEW_TRANSACTIONS_TOPIC: u32 = 0u32;
 pub const NEW_BLOCKS_TOPIC: u32 = 1u32;
 
+#[derive(Debug, Clone)]
+pub struct Gossip(pub Vec<Node>);
+
+impl gossip::Gossip for Gossip {
+    type Id = Id;
+    type Node = Node;
+
+    fn to_node_id(id: &Self::Id) -> gossip::Id {
+        gossip::Id(id.as_u128().to_be_bytes())
+    }
+
+    fn from_node_id(id: &gossip::Id) -> Self::Id {
+        Id::from(u128::from_be_bytes(id.0))
+    }
+
+    //fn to_nodes(&self) -> Box<Iterator<Item=Self::Node>> {
+    //    Box::new(self.0.into_iter())
+    //}
+
+    fn from_nodes(iter: Box<Iterator<Item = Self::Node>>) -> Self {
+        Gossip(iter.collect())
+    }
+}
+
+impl IntoIterator for Gossip {
+    type Item = Node;
+    type IntoIter = ::std::vec::IntoIter<Node>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
 /// object holding the P2pTopology of the Node
 #[derive(Clone)]
 pub struct P2pTopology {
     topology: Arc<RwLock<Topology>>,
 }
 
+impl property::Serialize for Gossip {
+    type Error = std::boxed::Box<bincode::ErrorKind>;
```

Comment:
Let's create a nice static error type wrapping the bincode stuff, and have it implement `std::error::Error`.

Reply:
This one is not resolved, would you like this to be the module's `Error` type?

Reply:
ha you are right, fixing!